File: Javascript/Sorting/Insertion_Sort.js





let arr=[5,3,2,4,1,9];

for(let i=0;i<arr.length-1;i++){
  let j=i;
  while(j>0&&arr[j-1]>arr[j]){
    let temp=arr[j];
    arr[j]=arr[j-1];
    arr[j-1]=temp;
     j--;
  }
 
}

console.log(arr)

File: C++/Sorting/Insertion_Sort.cpp


#include <iostream>
using namespace std;

void insertion_sort(int arr[],int n){
  for(int i=0;i<n-1;i++){
    int j=i;
    while(j>0 && arr[j-1]>arr[j]){
      int temp=arr[j-1];
      arr[j-1]=arr[j];
      arr[j]=temp;
      j--;
    }
  }
}

int main() 
{
  int n=6;
    int arr[n]={5,3,2,4,1,9};
    insertion_sort(arr,n);
    for(int i=0;i<n;i++){
      std::cout << arr[i] << std::endl;
    }
    
    return 0;
}